### ISSUE TITLE
Request for fix: AvoidImplicitelyRecompilingRegex String.split on external field

### ISSUE THREAD

**Issue by jborgers** [jborgers/PMD-jPinpoint-rules#53]
AvoidImplicitelyRecompilingRegex String.split fires on long regex, literals or by fields. 
External fields, from other classes, cannot be checked and currently fire a violation. This can be a false positive. We want to avoid false positives, so external fields should be assumed not to be a problem.
Example:
```
public class Try {
    private static final String SMALL = "-.";
    private static final String LARGE = "------.-";
    public static void foo(String query) {
        query.split(Constants.SEP); //assumed good, external
        query.split(SMALL); //good
        query.split(LARGE); //bad
    }
}
```

**Comment by jborgers** [comment]
fixed
